app/code/Magento/InventorySales/Plugin/CatalogInventory/StockManagement/ProcessRegisterProductsSalePlugin.php

```
@@ -94,61 +100,42 @@ public function __construct(
      * @throws LocalizedException
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function aroundRegisterProductsSale(StockManagement $subject, callable $proceed, $items, $websiteId = null)
-    {
+    public function aroundRegisterProductsSale(
+        StockManagement $subject,
+        callable $proceed,
+        $items,
+        $websiteId = null,
+        $quoteId = null
+    ) {
         if (empty($items)) {
             return [];
         }
         if (null === $websiteId) {
             throw new LocalizedException(__('$websiteId parameter is required'));
         }
+        if (null === $quoteId) {
+            //TODO: Do we need to throw exception?
+        }
```

I don't think we need to throw any exception when $quoteId is not passed, just create an empty metadata
This means that we need to have an ability to build a kind of `null object`  for SalesIvent
```
        $salesEvent = $this->salesEventFactory->create([
            'type' => 'none',
            'objectId' => 'null'
        ]);
```

---

TODO: use const for 'none' and think about `null`